### PR TITLE
feat(onyx-1225): add Shows for You home view section

### PR DESCRIPTION
**Pull request by nickskalkin**
This PR resolves [ONYX-1225]

### Description

This PR adds Shows for You rail to the new schema. We agreed to not add full Metaphysics support for this section (basically skip adding showsConnection). In this particular case it was very convenient to reuse existing component.

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1225]: https://artsyproduct.atlassian.net/browse/ONYX-1225?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/HomeView/Sections/Section.tsx [hunk 1]
```diff
@@ -39,6 +40,8 @@ export const Section: React.FC<{ section: SectionT }> = (props) => {
       return <FairsRailHomeViewSection section={section} />
     case "MarketingCollectionsRailHomeViewSection":
       return <MarketingCollectionsRailHomeViewSection section={section} />
+    case "ShowsRailHomeViewSection":
+      return <ShowsRailHomeViewSection section={section} />
```
Comment (anandaroop) [inline src/app/Scenes/HomeView/Sections/Section.tsx:44]: **thought (non-blocking):** [Related comment here](https://github.com/artsy/eigen/commit/8937f52830ace1b760a9157a5d5c698df3f655c7#r145756765), but no need to change this PR. If we agree on a naming convention we can update all of these at once.

File: src/app/Scenes/HomeView/Sections/ShowsRailHomeViewSection.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { ShowsRailHomeViewSection_section$key } from "__generated__/ShowsRailHomeViewSection_section.graphql"
+import { ShowsRailContainer } from "app/Scenes/Home/Components/ShowsRail"
+import { useFeatureFlag } from "app/utils/hooks/useFeatureFlag"
+import { graphql, useFragment } from "react-relay"
+
+interface ShowsRailHomeViewSectionProps {
+  section: ShowsRailHomeViewSection_section$key
+}
+
+export const ShowsRailHomeViewSection: React.FC<ShowsRailHomeViewSectionProps> = ({ section }) => {
```
Comment (anandaroop) [inline src/app/Scenes/HomeView/Sections/ShowsRailHomeViewSection.tsx:10]: **suggestion:** See Mounir's commit for an [example of a spec](https://github.com/artsy/eigen/commit/8937f52830ace1b760a9157a5d5c698df3f655c7#diff-242f62fb541f4be73b561c192e39788af681fe31f8e2af886fc0680a415c45de) for this kind of component.

File: src/app/Scenes/HomeView/Sections/ShowsRailHomeViewSection.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  const enableShowsForYouLocation = useFeatureFlag("AREnableShowsForYouLocation")
+  const data = useFragment(fragment, section)
+  const component = data.component
+
+  if (!component) return null
```
Comment (anandaroop) [inline src/app/Scenes/HomeView/Sections/ShowsRailHomeViewSection.tsx:15]: **question:** Is this early return necessary? Seems like if we want this component not to render, we would exclude it from the _server_ response.

File: src/app/Scenes/HomeView/Sections/ShowsRailHomeViewSection.tsx (hunk 2 above)
Comment (nickskalkin) [reply src/app/Scenes/HomeView/Sections/ShowsRailHomeViewSection.tsx:10]: thanks, done!

File: src/app/Scenes/HomeView/Sections/ShowsRailHomeViewSection.tsx (hunk 3 above)
Comment (nickskalkin) [reply src/app/Scenes/HomeView/Sections/ShowsRailHomeViewSection.tsx]: Removed it, not necessary

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f372cfca45fcb0421e02d50484abbfa01586513d
</p>

File: src/app/Scenes/HomeView/Sections/ShowsRailHomeViewSection.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+export const ShowsRailHomeViewSection: React.FC<ShowsRailHomeViewSectionProps> = ({ section }) => {
+  const enableShowsForYouLocation = useFeatureFlag("AREnableShowsForYouLocation")
+  const data = useFragment(fragment, section)
+  const component = data.component
+
+  return (
+    <ShowsRailContainer
+      title={component?.title || ""}
```
Comment (MounirDhahri) [inline src/app/Scenes/HomeView/Sections/ShowsRailHomeViewSection.tsx:17]:
```suggestion
      title={component?.title || "Shows"}
```

**Review by MounirDhahri — APPROVED** [review]
Nice! LGTM

**Pull request merged**